python/packages/kagent-langgraph/src/kagent/langgraph/_executor.py

```diff
@@ -110,24 +111,49 @@ async def _stream_graph_events(
         """Stream LangGraph events and convert them to A2A events."""
         task_result_aggregator = TaskResultAggregator()
 
+        # Track final state for interrupt detection
+        final_state: dict[str, Any] | None = None
+
+        # Track message IDs we've already sent to avoid duplicates
+        sent_message_ids: set[str] = set()
+
         # Stream events from the graph
         async for event in graph.astream(
             input_data,
             config,
-            stream_mode="updates",  # Stream the individual events
+            stream_mode="updates",
         ):
+            # Store final state
+            final_state = event
+
             # Convert LangGraph events to A2A events
             a2a_events = await _convert_langgraph_event_to_a2a(
-                event, context.task_id, context.context_id, self.app_name
+                event, context.task_id, context.context_id, self.app_name, sent_message_ids
             )
             for a2a_event in a2a_events:
                 task_result_aggregator.process_event(a2a_event)
                 await event_queue.enqueue_event(a2a_event)
 
+        # Check for interrupts after streaming completes
+        interrupt_detected = False
+        if final_state and final_state.get("__interrupt__"):
+            interrupt_detected = True
+            interrupt_data = final_state["__interrupt__"]
+            await self._handle_interrupt(
```

Comment:
Can't we just return here if an interrupt is detected? the 2 if blocks below don't execute if it is.